### ISSUE TITLE
Infinite recursion while encoding JSON

### ISSUE THREAD

**Issue by jfloff** [logtail/logtail-ruby-rack#12]
I've identified a bug in version `0.2.3` when used with Rails 6. When I execute my `rswag` specs, I encounter an issue of infinite recursion, seemingly during the JSON message encoding process. Below are the last few lines from my stacktrace:
```
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:154:in `block in as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:154:in `map'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:154:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:136:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:175:in `block in as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `each'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:58:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:175:in `block in as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `each'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:175:in `block in as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `each'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:174:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:58:in `as_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/json/encoding.rb:22:in `encode'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/json/encoding.rb:22:in `encode'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/core_ext/object/json.rb:43:in `to_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/log_entry.rb:88:in `to_json'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/logger.rb:99:in `call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/activesupport-6.1.7.6/lib/active_support/logger_thread_safe_level.rb:75:in `add'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/logger.rb:233:in `add'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/logger.rb:247:in `info'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-rack-0.2.3/lib/logtail-rack/http_events.rb:177:in `call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-rack-0.2.3/lib/logtail-rack/user_context.rb:74:in `call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-rails-0.2.4/lib/logtail-rails/session_context.rb:16:in `call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-rack-0.2.3/lib/logtail-rack/http_context.rb:23:in `block in call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/current_context.rb:120:in `with'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-0.1.12/lib/logtail/current_context.rb:44:in `with'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/logtail-rack-0.2.3/lib/logtail-rack/http_context.rb:22:in `call'
     # /heroku/ruby/bundle/ruby/3.1.4/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb:15:in `call'
```

When I lock `logtail-rack` to `0.2.2` version it works normally.

For reference, all my other versions of logtail gems are the most recent ones:
```
logtail (0.1.12)
      msgpack (~> 1.0)
logtail-rack (0.2.3)
      logtail (~> 0.1)
      rack (>= 1.2, < 4.0)
logtail-rails (0.2.4)
      actionpack (>= 5.0.0)
      activerecord (>= 5.0.0)
      logtail (~> 0.1)
      logtail-rack (~> 0.1)
      railties (>= 5.0.0)
```

**Comment by PetrHeinz** [comment]
Hi @jfloff, thanks for reporting this 🙌

I think I see where the problem is 👀 If I'm right, I should be able to fix it very soon.

**Comment by PetrHeinz** [comment]
Fixed and released as `v0.2.4` 🙂 Thanks again for reporting it with sufficient details to fix it swiftly!

**Comment by jfloff** [comment]
Can confirm it's fixed 👍  Thank you for the swift reply!
